crypto/ocsp/ocsp_integration_test.cc

```diff
@@ -181,31 +205,48 @@ TEST_P(OCSPIntegrationTest, AmazonTrustServices) {
   ASSERT_TRUE(SSL_library_init());
   bssl::UniquePtr<SSL_CTX> ssl_ctx(SSL_CTX_new(TLS_method()));
   ASSERT_TRUE(ssl_ctx);
+
+  // Set up trust store.
+  bssl::UniquePtr<X509_STORE> trust_store(SetupTrustStore());
+  ASSERT_TRUE(SSL_CTX_set1_verify_cert_store(ssl_ctx.get(), trust_store.get()));
   bssl::UniquePtr<BIO> bio(BIO_new_socket(sock, BIO_CLOSE));
   bssl::UniquePtr<SSL> ssl(SSL_new(ssl_ctx.get()));
+  // Set the host name, so we don't retrieve the wrong intended certificate
+  // chain.
+  SSL_set_tlsext_host_name(ssl.get(), t.url_host);
   SSL_set_bio(ssl.get(), bio.get(), bio.get());
   ASSERT_TRUE(SSL_connect(ssl.get()));
 
-  // Cert to be verified should be the first one in the chain, the Issuer of
-  // the Cert should be the one right after it.
-  STACK_OF(X509) *chain = SSL_get_peer_cert_chain(ssl.get());
-  X509 *certificate = sk_X509_value(chain, 0);
-  ASSERT_TRUE(certificate);
-  X509 *issuer = sk_X509_value(chain, 1);
+  STACK_OF(X509) *chain = SSL_get_peer_full_cert_chain(ssl.get());
+  EXPECT_EQ(SSL_get_verify_result(ssl.get()), X509_V_OK);
+  // Leaf to be verified should be the first one in the chain.
+  X509 *subject = sk_X509_value(chain, 0);
+  ASSERT_TRUE(subject);
```

Comment:
Is this getting the subject from the cert? Or is this still the entire x509 cert? 
```suggestion
  X509 *certificate = sk_X509_value(chain, 0);
  ASSERT_TRUE(certificate);
```